libsrc/hyperion/Hyperion.cpp

```diff
@@ -581,9 +586,32 @@ void Hyperion::update()
 
 		// feed smoothing in pause mode to maintain a smooth transistion back to smooth mode
 		if (_deviceSmooth->enabled() || _deviceSmooth->pause())
+		{
 			_deviceSmooth->setLedValues(_ledBuffer);
-
+		}
+		// Smoothing is disabled
 		if  (! _deviceSmooth->enabled())
+		{
 			emit ledDeviceData(_ledBuffer);
+		}
+	}
+	else
+	{
+		// LEDDevice is disabled and Smoothing is disabled
+
+		bool switchable = _ledDeviceWrapper->getActiveDevice()->isSwitchable();
+		// If LEDDevice is not switchable, simulate OFF-state providing a continous stream of BLACK
+		if (!switchable)
+		{
+			for (ColorRgb& color : _ledBuffer)
+			{
+				color = ColorRgb::BLACK;
+			}
+			emit ledDeviceData(_ledBuffer);
+		}
+//		else
+//		{
+//			// Device has been switched off
+//		}
```

Comment:
Dieser Bereicht ist an sich überflüssig. Nicht schaltbare Geräte sollte mal nicht mit Schwarzwerten fluten.
Sie erhalten einmal den Schwarzwert [hier](https://github.com/hyperion-project/hyperion.ng/blob/986b9dc979116c45daccfc5286912190af8b6d31/libsrc/leddevice/LedDevice.cpp#L114).
Welchen Grund hattest du denn um diese Passage ein zu führen?

Reply:
@Paulchen-Panther 

> Dieser Bereicht ist an sich überflüssig. Nicht schaltbare Geräte sollte mal nicht mit Schwarzwerten fluten.
> Sie erhalten einmal den Schwarzwert [hier](https://github.com/hyperion-project/hyperion.ng/blob/986b9dc979116c45daccfc5286912190af8b6d31/libsrc/leddevice/LedDevice.cpp#L114).
> Welchen Grund hattest du denn um diese Passage ein zu führen?

Gemäß @brindosch 's Kommentar 

>  people claimed that their "led stripes" might go on if there is no data stream.

ging hier um die Konsistenz. Wenn Smoothing und das Device jeweils  disabled sind, wird dieser Pfad durchlaufen. Damit auch in diesem Fall (bei einem non-switchable Device) immer Schwarz ausgegeben wird. Wenn der Code nicht da ist findet keine Ausgabe statt.

Reply:
Ich betreibe z.b. einen Arduino (serial) und einen Fadecandy (network) mit ws2812b. Diese gehen nicht von alleine an. Und ich würde gerne den Arduino bzw. mein Fadcandy Netzwerk nicht mit Schwarzwerten fluten. Lieber eine optionale auswahl ob dies auch gewünscht ist. Oder wie Brindosch meinte
> I'm not sure. Bulbs should have no problem with this. But everything that ends in a "custom" led stripe might go crazy **after longer idle time**
dann nur in bestimmten Zeitabständen den Schwarzwert senden..

Ach ja, bevor ich es vergesse. Die bool Zuweisung in Zeile 602 sollte, wenn es dochdrin bleibt nicht mit in die update Routine. Da jede Instanz nur ein LED Device steuern kann, sollte sich das nur in der handleSettingsUpdate() ändern.

Reply:
> Ich betreibe z.b. einen Arduino (serial) und einen Fadecandy (network) mit ws2812b. Diese gehen nicht von alleine an. Und ich würde gerne den Arduino bzw. mein Fadcandy Netzwerk nicht mit Schwarzwerten fluten. Lieber eine optionale auswahl ob dies auch gewünscht ist. Oder wie Brindosch meinte
> 
> > I'm not sure. Bulbs should have no problem with this. But everything that ends in a "custom" led stripe might go crazy **after longer idle time**
> > dann nur in bestimmten Zeitabständen den Schwarzwert senden..
> 
> Ach ja, bevor ich es vergesse. Die bool Zuweisung in Zeile 602 sollte, wenn es dochdrin bleibt nicht mit in die update Routine. Da jede Instanz nur ein LED Device steuern kann, sollte sich das nur in der handleSettingsUpdate() ändern.

Wenn diese LedDevices richtig schaltbar sind, dann spricht ja nichts dagegen, ihnen das Property Switchable per se zu geben (so wie ich im Nanoleaf Device). 
Oder wir entscheiden uns, dass der Default "keine Schwarz-Simulation" ist.

Reply:
Die sind nicht schaltbar. Die brauchen nur einmal den schwarzwert.

Reply:
> Oder wir entscheiden uns, dass der Default "keine Schwarz-Simulation" ist.

👍

Reply:
> Die bool Anweisung ist dafür da, um zu schauen ob das Device switchable ist.
> Ohne diese Anweisung würde die Switchability nicht berücksichtigt werde...

Da reicht aber einmal die abfrage und nicht jedes mal

Reply:
Weches  handleSettingsUpdate() meinst Du, das von Hyperion oder LinearColorSmoothing?

Reply:
Oder wir entscheiden uns, dass der Default "keine Schwarz-Simulation" ist.

Dass heißt, ich setzt den Default Wert von Switchable im Led Device auf True.
Dann können wir auch den Abschnitt hier erstmal löschen/auskommentieren...

Reply:
> Weches handleSettingsUpdate() meinst Du, das von Hyperion oder LinearColorSmoothing?

Natürlich von Hyperion oder wo hast du du [diese Zeile hier platziert](https://github.com/hyperion-project/hyperion.ng/pull/629/files/986b9dc979116c45daccfc5286912190af8b6d31#diff-6266746bda3a23d2eecfdb108923208bR602)?

Reply:
Ok. Muss noch mal schauen, ob handleSettingsUpdate aufgerufen wird, wenn ich die LED ausschalte....
Ok. Dann würde Hyperion auch eine private "switchable" Variable bekommen und die würde dann im handleSettingsUpdate gesetzt und im Update abgefragt.

Ist es das, was Du meinst?

Reply:
> Ist es das, was Du meinst?

NEIN

Reply:
Das kann doch nicht so schwer sein. Sorry.
Du fragst jedes mal, wenn deine LEDs neue Daten bekommen in Hyperion::update() ab ob das Device switchable ist? Warum zum Geier?
Das ändern sich doch im laufenden betrieb nicht! Da muss schon handleSettingsUpdate() aufgerufen werden bevor sich was am Typ deines Device etwas ändert.

Reply:
Dann versuche ich es nochmal anders....
Unter der Annahme, dass wenn ein LedDevice ausgeschaltet ist ein kontinuierlicher Schwarz-Stream ausgegeben werden soll, muss die Update Routine jetzt "Schwarz" emittieren.
Allerdings soll das nicht für switchable Devices passieren. Die können ausgeschaltet werden, und brauchen diesen Stream nicht.

Der Code sag doch nur

update ()
...
if Leddevice **enabled**
emit ledDeviceData (mit oder ohne smotthing)

else  - **LedDevice ist ausgeschaltet**

Wenn nicht switchable, dann simuliere kontinuierlich Schwarz
emittiere ledDeviceData in Schwarz
 
Damit "**Wenn nicht switchable**" functioniert, muß das switchable Property der LedDevices gelesen werden (oder man hat es als member der hyperion Klasse).
Ich weiß nicht wie Du sonst die beiden Fälle unterscheiden willst?

Reply:
Wenn wir bei LedDevice = aus, keinen kontinuierlichen Schwarz-Stream erzeugen, dann kann der gesamte Teil gelöscht werden.... ansonsten muß das switchable Property berücksichtigt werden

Reply:
Es geht doch nur darum das du nicht die update schleife übermäßig vollmüllst. 
Um so länger die wird um so mehr geht die Performance bei schwächeren Geräten runter

Reply:
Schon verstanden... Darum habe ich es auch im Update implementiert und nicht erst im LedDevice write ... :-)

Nachdem dieser Call eh nur im "Exception" Fall (Device und Smotthing sind aus) aufgerufen wird, sollte die User-Experience überschaubar sein...

Reply:
Ich schlage vor wir kommentieren den Code erstmal aus und falls es im Feld nach der Beta Version kontinuierliche "Schwarz" Simulation braucht, können wir den Code aktivieren.

Ich checke nochmal, ob bei Default switchable=true für alle Devices noch ein letzter Wert von Schwarz geschickt wird....

Reply:
Ok. Danke